### DISCUSSION TITLE
Error Regarding a Specific Function in .NET Library

### DISCUSSION THREAD

**Discussion by Nao050101** [pythonnet/pythonnet#2158] [Q&A] [answered]
I tried to call a function in a class library using .NET 5.0 framework using pythonnet, but I encountered an error stating that the following function is missing.
Please tell me the solution.

### The Python code from the caller is as follows.
try:
    import clr
    lib_path=".dll/testdll/TestLib
    clr.AddReference(lib_path)
    from TestLib import TestClass
except Exception as e: 
    print(e)   **"""System.Runtime, Version4.0.0.0, Culture＝neutral, System.Threading.Tasks.Task'1' LoadError**

test=TestClass()
test.test()

### I will provide a simplified version of the .NET class library that you are calling as mentioned below.

public class TestClass
{         
 

        //OK function
        public static List<string> TestNormal()
        {
                  var testList=new List<string>();
                  testList.Add("ABC")
                  return testList
        }

      //NG function
       public static async Task<List<string>> TestAsync()
       {
                  var testList=new List<string>();
                  testList.Add("ABC")
                  return testList
        }

        public bool test()
        {
                var task =TestAsync()
                task.Wait();
               var result=task.Result;
        }
}

**Comment by filmor** [comment, marked as answer]
Please read the documentation. `import clr` will load the default runtime, which for compatibility reasons is Mono on Linux and macOS and .NET Framework on Windows which can't run .NET Core DLLs.

**Reply by Nao050101** [reply]
Thank you, I appreciate your help as it has brought me closer to the answer.

I understand that it does not support .NET Core.

import clr

However, is there any way to call .NET Core libraries from Python?

**Reply by Nao050101** [reply]
After adding the following lines, it started working. Thank you very much.

from pythonnet import load
load("coreclr")
import clr

**Comment by Nao050101** [comment]
Thankyou.
